localstack-core/localstack/services/lambda_/invocation/execution_environment.py

```diff
@@ -343,11 +341,22 @@ def get_prefixed_logs(self) -> str:
 
     def invoke(self, invocation: Invocation) -> InvocationResult:
         assert self.status == RuntimeStatus.RUNNING
+        # Async/event invokes might miss an aws_trace_header, then we need to create a new root trace id.
+        aws_trace_header = (
+            invocation.trace_context.get("aws_trace_header") or TraceHeader().ensure_root_exists()
+        )
+        # The Lambda RIE requires a full tracing header including Root, Parent, and Samples. Otherwise, tracing fails
+        # with the warning "Subsegment ## handler discarded due to Lambda worker still initializing"
+        aws_trace_header.ensure_sampled_exists()
```

Comment:
so if I understand that right, for lambda it's always `Sampled=1` while other services will by default get a `Sampled=0`? :thinking:

Reply:
no, the sampled field is missing altogether (i.e., not present rather than sampled=0) for other services if no incoming tracing header is present with the sampled field.

That's not ideal, but this depends on the (service-dependent) sampling strategies and other services. We're lacking parity tests here, so idk what's the best behavior. Just leaving it open for now ...